src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/TextMapPropagator.cs

```diff
@@ -0,0 +1,139 @@
+
+        /// <summary>
+        /// The predefined propagation fields
+        /// </summary>
+        /// <returns>Returns list of fields that will be used by the TextMapPropagator.</returns>
+        public abstract IReadOnlyCollection<string> Fields { get; }
+
+        /// <summary>
+        /// Injects the trace values stroed in the <see cref="Activity"/> object into a carrier. For example, into the headers of an HTTP request.
+        /// </summary>
+        /// <param name="activity">The Activity object has the trace context to inject to the carrier.</param>
+        /// <param name="carrier">Carrier is the medium used by the propagators to write values to.</param>
+        /// <param name="setter">The callback will be invoked to set a propagation key/value pair to the carrier.</param>
+        public abstract void Inject(Activity activity, object carrier, PropagatorSetterCallback setter);
+
+        /// <summary>
+        /// Extracts the value from an incoming request represented by the carrier. For example, from the headers of an HTTP request.
+        /// </summary>
+        /// <param name="carrier">Carrier is the medium used by the propagators to read values from.</param>
+        /// <param name="getter">The callback will be invoked to get the propagation trace Id and trace state from carrier.</param>
+        /// <param name="traceId">The extracted trace Id from the carrier.</param>
+        /// <param name="traceState">The extracted trace state from the carrier.</param>
+        public abstract void ExtractTraceIdAndState(object carrier, PropagatorGetterCallback getter, out string? traceId, out string? traceState);
+
+        /// <summary>
+        /// Extracts the baggage key-value pairs list from an incoming request represented by the carrier. For example, from the headers of an HTTP request.
+        /// </summary>
+        /// <param name="carrier">Carrier is the medium used by the propagators to read values from.</param>
+        /// <param name="getter">The callback will be invoked to get the propagation baggage list from carrier.</param>
+        /// <returns>Returns the extracted key-value pair list from teh carrier.</returns>
+        public abstract IEnumerable<KeyValuePair<string, string?>>? ExtractBaggage(object carrier, PropagatorGetterCallback getter);
+
+        /// <summary>
+        /// Get or set the process wide propagator object which used as the current selected propagator.
+        /// </summary>
+        public static TextMapPropagator Current
+        {
+            get
+            {
+                Debug.Assert(s_current is not null);
+                return s_current;
+            }
+
+            set
+            {
+                if (value is not null)
+                {
+                    s_current = value;
+                }
+            }
+        }
+
+        /// <summary>
+        /// returns the default propagator object which Current property will be initialized with.
```

Comment:
Can we have this comment describe the behavior of the current default, and change it if we change the default in the future?